### PR TITLE
Fix the shuffling of labels

### PR DESCRIPTION
**Pull request by arthurPignet**
The shuffle of labels for a shuffled partner doesn't work.

### CODE REVIEW

File: mplc/partner.py [hunk 1]
```diff
@@ -54,19 +54,6 @@ def shuffle_labels(self, proportion_shuffled):
...
-            label = self.y_train[i]
-            # Suffle the label
-            new_label = shuffle(label)
-
-            # Force the label to be different
-            while np.all(new_label == label):
-                new_label = shuffle(label)
-
-            self.y_train[i] = new_label
+        np.random.shuffle(self.y_train[:n])
```
Comment (Thomas-Galtier) [inline mplc/partner.py:59]: This won't work if your partner as only one class.

File: mplc/partner.py [hunk 2]
```diff
@@ -54,19 +54,7 @@ def shuffle_labels(self, proportion_shuffled):
...
-            label = self.y_train[i]
-            # Suffle the label
-            new_label = shuffle(label)
-
-            # Force the label to be different
-            while np.all(new_label == label):
-                new_label = shuffle(label)
-
-            self.y_train[i] = new_label
+        for label in self.y_train[:n]:
```
Comment (RomainGoussault) [inline mplc/partner.py:59]: We want to shuffle only the first n labels ? 
I think it makes more sense to randomly choose n labels among len(y_train) and to shuffle them.

File: mplc/partner.py [hunk 3]
```diff
@@ -54,19 +54,7 @@ def shuffle_labels(self, proportion_shuffled):
             raise ValueError(
                 f"The proportion of labels to corrupted was {proportion_shuffled} but it must be between 0 and 1."
             )
-
         # Select the indices where the label will be shuffled
```
Comment (RomainGoussault) [inline mplc/partner.py:57]: We can delete this comment I guess

**Review by RomainGoussault — CHANGES_REQUESTED** [review]

File: mplc/partner.py (hunk 2 above)
Comment (arthurPignet) [reply mplc/partner.py]: I though labels were shuffled during the data splittin between partners, and before each minibatch split. Nevertheless, in advanced split, it is not obvious that the data are shuffled. I changed that

**Review by bowni — APPROVED** [review]

**Review by natct10 — APPROVED** [review]

**Review by RomainGoussault — APPROVED** [review]
👍

**Pull request merged**